fe/fe-core/src/main/java/org/apache/doris/external/jdbc/JdbcClient.java

```diff
@@ -0,0 +1,422 @@
+            //          `null` means get all schema;
+            //          Can contain single-character wildcards ("_"), or multi-character wildcards ("%")
+            // tableNamePattern - table name
+            //                    Can contain single-character wildcards ("_"), or multi-character wildcards ("%")
+            // columnNamePattern - column name, `null` means get all columns
+            //                     Can contain single-character wildcards ("_"), or multi-character wildcards ("%")
+            rs = databaseMetaData.getColumns(dbName, null, tableName, null);
+            while (rs.next()) {
+                JdbcFieldSchema field = new JdbcFieldSchema();
+                field.setColumnName(rs.getString("COLUMN_NAME"));
+                field.setDataType(rs.getInt("DATA_TYPE"));
+                field.setDataTypeName(rs.getString("TYPE_NAME"));
+                field.setColumnSize(rs.getInt("COLUMN_SIZE"));
+                field.setDecimalDigits(rs.getInt("DECIMAL_DIGITS"));
+                field.setNumPrecRadix(rs.getInt("NUM_PREC_RADIX"));
+                field.setNullAble(rs.getInt("NULLABLE"));
+                field.setRemarks(rs.getString("REMARKS"));
+                field.setCharOctetLength(rs.getInt("CHAR_OCTET_LENGTH"));
+                tableSchema.add(field);
+            }
+        } catch (SQLException e) {
+            throw new JdbcClientException("failed to get table name list from jdbc for table %s", e, tableName);
+        } finally {
+            close(rs, conn);
+        }
+        return tableSchema;
+    }
+
+    public Type jdbcTypeToDoris(JdbcFieldSchema fieldSchema) {
+        switch (dbType) {
+            case MYSQL:
+                return mysqlTypeToDoris(fieldSchema);
+            default:
+                throw new JdbcClientException("Unknown database type");
+        }
+    }
+
+    public Type mysqlTypeToDoris(JdbcFieldSchema fieldSchema) {
+        // For mysql type: "INT UNSIGNED", we use split(" ")[0] to get "INT"
+        String[] typeFields = fieldSchema.getDataTypeName().split(" ");
+        String mysqlType = typeFields[0];
+        switch (mysqlType) {
+            case "BOOLEAN":
+                return Type.BOOLEAN;
+            case "TINYINT":
+                return Type.TINYINT;
+            case "SMALLINT":
+                return Type.SMALLINT;
+            case "MEDIUMINT":
+            case "INT":
+                return Type.INT;
+            case "BIGINT":
+                return Type.BIGINT;
+            case "DATE":
+                return ScalarType.getDefaultDateType(Type.DATE);
+            case "TIMESTAMP":
+                return ScalarType.getDefaultDateType(Type.DATETIME);
+            case "DATETIME":
+                return ScalarType.createCharType(8);
+            case "TIME":
+                return ScalarType.createCharType(3);
+            case "YEAR":
+                return ScalarType.createCharType(1);
```

Comment:
For YEAR/TIME/TIMESTAMP, maybe  we should use integer. Better to test them.

Reply:
After test:
for YEAR, we can use doris MEDIUMINT data type.
for TIME, we can use doris STRING data type.
for TIMESTAMP, we can use doris DATETIME data type.